examples/hotel-bookings/src/index.md

```diff
@@ -0,0 +1,264 @@
+  return Plot.plot({
+    width,
+    height: height - 30,
+    x: { label: "Booking lead time (days)" },
+    y: { grid: true },
+    color: { legend: true },
+    marks: [
+      Plot.ruleY([0]),
+      Plot.axisX({ ticks: 5 }),
+      Plot.axisY({ ticks: 5 }),
+      Plot.rectY(
+        bookingsByMarketSegment,
+        Plot.binX(
+          { y: "count" },
+          { x: "LeadTime", interval: 20, fill: "season" }
+        )
+      ),
+    ],
+  });
+}
+```
+
+```js
+// Daily rate stacked histogram
+function dailyRateChart(width, height) {
+  return Plot.plot({
+    width,
+    height: height - 30,
+    x: { label: "Daily rate" },
+    y: { grid: true },
+    color: { legend: true },
+    marks: [
+      Plot.ruleY([0]),
+      Plot.axisX({ ticks: 5 }),
+      Plot.axisY({ ticks: 5 }),
+      Plot.rectY(
+        bookingsByMarketSegment,
+        Plot.binX(
+          { y: "count" },
+          { x: "ADR", fill: "#21C6A8", interval: 20, fill: "season", tip: true}
+        )
+      ),
+    ],
+  });
+}
+```
+
+**Data source:** Antonio et al (2021). Hotel booking demand datasets. Data in Brief (22): 41-49. https://doi.org/10.1016/j.dib.2018.11.126
+
+**Credit:** Donut charts reuse code from [Mike Bostock's _Donut chart_ notebook](https://observablehq.com/@d3/donut-chart/2). Big number boxes are adapted from [Paul Buffa’s _Walmart growth dashboard_](https://observablehq.com/d/a156c1245672f9bd).
+
+```js
+// Data wrangling for market segmented charts
+const bookingsByMarketSegment = hotelData.filter(
+  (d) =>
+    d.MarketSegment == pickMarketSegment && d.MarketSegment != "Complementary"
+);
+
+// By booking country
+const bookingCountry = d3
+  .flatRollup(
+    bookingsByMarketSegment,
+    (d) => d.length,
+    (v) => v.Country
+  )
+  .map(([name, value]) => ({ name, value }))
+  .sort((a, b) => d3.descending(a.value, b.value));
+
+const bookingCountryTopN = bookingCountry.slice(0, 5);
+
+const bookingCountryOther = {
+  name: "Other",
+  value: d3.sum(
+    bookingCountry.slice(5 - bookingCountry.length),
+    (d) => d.value
+  ),
+};
+
+const byCountry = bookingCountryTopN.concat(bookingCountryOther);
+
+//Booking outcome (cancelled or not cancelled)
+const byBookingOutcome = d3
+  .flatRollup(
+    bookingsByMarketSegment,
+    (d) => d.length,
+    (d) => d.IsCanceled
+  )
+  .map(([name, value]) => ({ name, value }))
+  .sort((a, b) => d3.descending(a.value, b.value));
+
+// Bookings by Room Type Data
+const byRoomType = d3
+  .flatRollup(
+    bookingsByMarketSegment,
+    (d) => d.length,
+    (d) => d.ReservedRoomType
+  )
+  .map(([name, value]) => ({ name, value }))
+  .sort((a, b) => d3.descending(a.value, b.value));
+
+const bookingSeason = d3
+  .flatRollup(
+    bookingsByMarketSegment,
+    (d) => d.length,
+    (v) => v.season
+  )
+  .map(([name, value]) => ({ name, value }))
+  .sort((a, b) => d3.descending(a.value, b.value));
+```
+
+```js
+const cleanTable = bookingsByMarketSegment.map(d => ({
```

Comment:
cleanTable is mostly to rename columns? for this we should use the _header_ option of Inputs.table. And total nights could be computed upstream when loading the data.

Reply:
Moved into Inputs.table options